matrix-sdk-android/src/test/java/org/matrix/android/sdk/internal/session/room/EditValidationTest.kt

```diff
@@ -0,0 +1,366 @@
+/*
+ * Copyright (c) 2022 The Matrix.org Foundation C.I.C.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.matrix.android.sdk.internal.session.room
+
+import io.mockk.every
+import io.mockk.mockk
+import org.amshove.kluent.shouldBeInstanceOf
+import org.junit.Test
+import org.matrix.android.sdk.api.session.crypto.model.CryptoDeviceInfo
+import org.matrix.android.sdk.api.session.crypto.model.OlmDecryptionResult
+import org.matrix.android.sdk.api.session.events.model.Event
+import org.matrix.android.sdk.api.session.events.model.EventType
+import org.matrix.android.sdk.internal.crypto.store.IMXCryptoStore
+
+class EditValidationTest {
+
+    private val mockTextEvent = Event(
+            type = EventType.MESSAGE,
+            eventId = "\$WX8WlNC2reiXrwHIA_CQHmU_pSR-jhOA2xKPRcJN9wQ",
+            roomId = "!GXKhWsRwiWWvbQDBpe:example.com",
+            content = mapOf(
+                    "body" to "some message",
+                    "msgtype" to "m.text"
+            ),
+            originServerTs = 1000,
+            senderId = "@alice:example.com",
+    )
+
+    private val mockEdit = Event(
+            type = EventType.MESSAGE,
+            eventId = "\$-SF7RWLPzRzCbHqK3ZAhIrX5Auh3B2lS5AqJiypt1p0",
+            roomId = "!GXKhWsRwiWWvbQDBpe:example.com",
+            content = mapOf(
+                    "body" to "* some message edited",
+                    "msgtype" to "m.text",
+                    "m.new_content" to mapOf(
+                            "body" to "some message edited",
+                            "msgtype" to "m.text"
+                    ),
+                    "m.relates_to" to mapOf(
+                            "rel_type" to "m.replace",
+                            "event_id" to mockTextEvent.eventId
+                    )
+            ),
+            originServerTs = 2000,
+            senderId = "@alice:example.com",
+    )
+
+    @Test
+    fun `edit should be valid`() {
+        val mockCryptoStore = mockk<IMXCryptoStore>()
+        val validator = EventEditValidator(mockCryptoStore)
+
+        validator
+                .validateEdit(mockTextEvent, mockEdit) shouldBeInstanceOf EventEditValidator.EditValidity.Valid::class
+    }
+
+    @Test
+    fun `original event and replacement event must have the same sender`() {
+        val mockCryptoStore = mockk<IMXCryptoStore>()
```

Comment:
Like for other fake interfaces, I think we should create a new reusable Fake class for `IMXCryptoStore`.